resources/test/tests/worker-dedicated.html

```diff
@@ -30,8 +30,8 @@ <h1>Dedicated Web Worker Tests</h1>
 {
   "summarized_status": {
     "status_string": "ERROR",
-    "message": "Error: This failure is expected.",
-    "stack": "(implementation-defined)"
+    "message": "Error in remote https://web-platform.test:9003/resources/test/tests/worker-error.js: Error: This failure is expected.",
```

Comment:
The `remote_done` function severs the connection that a "remote" environment
has with its parent. It is intended to be invoked when a remote test completes.

- In `master` today, `remote_done` is invoked before `remote_error`. When
  `remote_error` is later invoked, it delegates to `remote_done`. However,
  because that method has already been invoked, this causes an internal
  exception which interrupts further execution of the function body.
  `remote_error` is designed to invoke `preventDefault` on the uncaught
  exception event, but this never happens due to the runtime exception. That
  means the error "bubbles up" to the parent context. `testharness.js` (the
  instance running in the parent context) reacts to the uncaught error by
  updating the status accordingly.

  This is why the current failure message inaccurately describes an error that
  originated in the parent context.
- With this patch applied, delegation to `remote_done` in `remote_error` has
  been refactored, so in the internal exception does not occur in this test,
  and the subsequent call to `preventDefault` proceeds as intended. This means
  the parent context does not receive an "uncaught error" event. Instead, it
  accurately reports the error with the text, "Error in remote [...]".

Reply:
It was the "web-platform.test:9003" bit that worried me, not the change in behavior itself, although I did not understand the cause of that either. The new behavior seems fine, and if the port will always be 9003 that's also fine. Can you confirm that it passes if run locally, not just on Travis?

Reply:
> Can you confirm that it passes if run locally, not just on Travis?

Every time :)